### PR TITLE
Added examples of /etc/security/limits.conf changes

### PR DESCRIPTION
**Pull request by sarwarbhuiyan**
- Added sections for memlock, nproc, and as
- changed links in bootstrap configs to point to limits.conf
  section

### CODE REVIEW

**Comment by sarwarbhuiyan** [comment]
Fixes 23274

File: docs/reference/setup/bootstrap-checks.asciidoc [hunk 1]
```diff
@@ -90,7 +91,7 @@ that the Elasticsearch process has the rights to create enough threads
 under normal use. This check is enforced only on Linux. If you are on
 Linux, to pass the maximum number of threads check, you must configure
 your system to allow the Elasticsearch process the ability to create at
-least 2048 threads. This can be done via `/etc/security/limits.conf`
+least 2048 threads. This can be done via <limits.conf,`/etc/security/limits.conf`>>
```
Comment (clintongormley) [inline docs/reference/setup/bootstrap-checks.asciidoc]: missing `<`

File: docs/reference/setup/bootstrap-checks.asciidoc [hunk 2]
```diff
@@ -105,7 +106,7 @@ maximum size virtual memory check enforces that the Elasticsearch
 process has unlimited address space and is enforced only on Linux. To
 pass the maximum size virtual memory check, you must configure your
 system to allow the Elasticsearch process the ability to have unlimited
-address space. This can be done via `/etc/security/limits.conf` using
+address space. This can be done via <limits.conf,`/etc/security/limits.conf`>> using
```
Comment (clintongormley) [inline docs/reference/setup/bootstrap-checks.asciidoc]: missing `<`

**Review by clintongormley — COMMENTED** [review]
I'll leave it to @jasontedor to comment on the correctness of the settings

File: docs/reference/setup/sysconfig/configuring.asciidoc [hunk 3]
```diff
@@ -50,7 +52,35 @@ the `limits.conf` file:
 elasticsearch  -  nofile  65536
 --------------------------------
 
-This change will only take effect the next time the `elasticsearch` user opens
+We recommend setting <<bootstrap.memory_lock,`bootstrap.memory_lock`>>.
```
Comment (jasontedor) [inline docs/reference/setup/sysconfig/configuring.asciidoc]: This is our last recommended way of minimizing the impact of swap. I think that we should remove the part about recommending here and only say that if you want to enable Elasticsearch to lock memory, you must permit the `elasticsearch` user to lock an unlimited amount of memory.

File: docs/reference/setup/sysconfig/configuring.asciidoc [hunk 4]
```diff
@@ -50,7 +52,35 @@ the `limits.conf` file:
...
+In order for Elasticsearch to be able to lock the heap in memory through 
+mlockall and avoid swapping, add the following line to the `limits.conf`
```
Comment (jasontedor) [inline docs/reference/setup/sysconfig/configuring.asciidoc]: These days the recommended approach would be to create a file in `/etc/security/limits.d/` (e.g., `elasticsearch.conf`). I think that while we are here, we should go the extra mile and change these recommendations to be centered around this.

**Review by jasontedor — CHANGES_REQUESTED** [review]
Thanks @sarwarbhuiyan, I left some comments.

**Comment by sarwarbhuiyan** [comment]
@jasontedor left some changes.

**Comment by rjernst** [comment]
@jasontedor Are you happy with the updates that were made? Is this worth merging?

**Comment by dakrone** [comment]
ping @jasontedor since this is waiting on your +1 or (currently standing) -1

**Comment by jasontedor** [comment]
Closing as the CLA is not signed. We can reopen if this changes.

**Pull request closed without merging**